File: Algorithms/Ruby/maximum-depth-of-binary-tree.rb
# Definition for a binary tree node.
# class TreeNode
#     attr_accessor :val, :left, :right
#     def initialize(val)
#         @val = val
#         @left, @right = nil, nil
#     end
# end

# @param {TreeNode} root
# @return {Integer}

def maximum_depth_of_binary_tree(treeNode)
  treeNode.nil? ? 0 : 1 + [maximum_depth_of_binary_tree(treeNode.left), maximum_depth_of_binary_tree(treeNode.right)].max
end

File: Algorithms/maximum-depth-of-binary-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 * 
 * @param {TreeNode} root
 * @return {number}
 */

var maximumDepthOfBinaryTree = function(treeNode) {
  return treeNode === null ?
    0 :
    1 + Math.max(maximumDepthOfBinaryTree(treeNode.left), maximumDepthOfBinaryTree(treeNode.right));
};
